tuned/ppd/controller.py

```diff
@@ -174,16 +200,22 @@ def set_active_profile(self, profile):
             raise dbus.exceptions.DBusException("Invalid profile '%s'" % profile)
         self._base_profile = profile
         self._profile_holds.clear()
+        if self._on_battery:
+            profile = "%s-battery" % profile
         self.switch_profile(profile)
 
     @exports.property_getter("ActiveProfile")
     def get_active_profile(self):
-        return self.active_profile()
+        return self.active_profile().replace("-battery", "")
 
     @exports.property_getter("Profiles")
     def get_profiles(self):
+        profiles = []
+        for profile in ["power-saver", "balanced", "performance"]:
+            if profile in self._config.ppd_to_tuned:
+                profiles.append(profile)
         return dbus.Array(
-            [{"Profile": profile, "Driver": DRIVER} for profile in self._config.ppd_to_tuned.keys()],
+            [{"Profile": profile, "Driver": DRIVER} for profile in profiles],
```

Comment:
This, for instance, wouldn't be necessary if using a separate config section for the "on-battery" profiles.